### PR TITLE
Tests for abortSignal.throwIfAborted();

### PR DESCRIPTION
**Pull request by jasnell**
Refs: https://github.com/whatwg/dom/pull/1034

### CODE REVIEW

File: dom/abort/event.any.js [hunk 1]
```diff
@@ -139,4 +139,22 @@ test(t => {
...
+test(t => {
+  const reason = new Error('boom');
+  const signal = AbortSignal.abort(reason);
+  assert_true(signal.aborted);
+  try {
+    signal.throwIfAborted();
+    throw new Error("failed to throw when expected");
+  } catch (err) {
+    assert_equals(err, reason);
+  }
```
Comment (annevk) [inline dom/abort/event.any.js:151]: You want https://web-platform-tests.org/writing-tests/testharness-api.html#assert_throws_exactly.
```suggestion
  assert_throws_exactly(reason, () => signal.throwIfAborted());
```

**Review by annevk — COMMENTED** [review]
Maybe also add a test with a primitive value as reason?

**Review by domenic — APPROVED** [review]
I feel like this file is no longer just about the event, so really should be split up into multiple files, but that's a preexisting issue so let's merge this first and then split up or rename later if we care.

**Comment by domenic** [comment]
Oh, I missed @annevk's suggestion to do a primitive value as the reason. I don't think it's really necessary but I lean toward more coverage rather than less so let's do it, before we merge.

**Comment by jasnell** [comment]
Added the additional primitive check

**Pull request merged**